src/java.base/share/classes/java/lang/Float.java

```diff
@@ -975,6 +976,206 @@ public static int floatToIntBits(float value) {
+     * mode}.
+     *
+     * Special cases:
+     * <ul>
+     * <li> If the argument is zero, the result is a zero with the
+     * same sign as the argument.
+     * <li> If the argument is infinite, the result is an infinity
+     * with the same sign as the argument.
+     * <li> If the argument is a NaN, the result is a NaN.
+     * </ul>
+     *
+     * The <a href="#binary16Format">binary16 format</a> is discussed in
+     * more detail in the {@link #binary16AsShortBitsToFloat} method.
+     *
+     * @apiNote
+     * This method corresponds to the convertFormat operation defined
+     * in IEEE 754 from the binary32 format to the binary16 format.
+     * The operation of this method is analogous to a primitive
+     * narrowing conversion (JLS {@jls 5.1.3}).
+     *
+     * @param f the {@code float} value to convert to binary16
+     * @since 20
+     */
+    // @IntrinsicCandidate
+    public static short floatToBinary16AsShortBits(float f) {
+        if (Float.isNaN(f)) {
+            // Arbitrary binary16 NaN value; could try to preserve the
+            // sign and some bits of the NaN significand.
+            return (short)0x7fff;
+        }
+
+        float abs_f = Math.abs(f);
+        int doppel = Float.floatToRawIntBits(f);
+        short sign_bit = (short)((doppel >> 16) & 0x8000);
+
+        // The overflow threshold is binary16 MAX_VALUE + 1/2 ulp
+        if (abs_f > (65504.0f + 16.0f) ) {
+            return (short)(sign_bit | 0x7c00); // Positive or negative infinity
+        } else {
+            // Smallest magnitude nonzero representable binary16 value
+            // is equal to 0x1.0p-24; half-way and smaller rounds to zero.
+            if (abs_f <= 0x1.0p-25f) { // Covers float zeros and subnormals.
+                return sign_bit; // Positive or negative zero
+            }
+
+            // Dealing with finite values in exponent range of
+            // binary16 (when rounding is done, could still round up)
+            int exp = Math.getExponent(f);
+            assert -25 <= exp && exp <= 15;
+            short signif_bits;
+
+            if (exp <= -15) { // scale down to float subnormal range to do rounding
+                // Use a float multiply to compute the correct
+                // trailing significand bits for a binary16 subnormal.
+                //
+                // The exponent range of normalized binary16 subnormal
+                // values is [-15, -24]. The exponent range of float
+                // subnormals is [-140, -149]. Multiply abs_f down by
+                // 2^(-125) -- since (-125 = -149 - (-24)) -- so that
+                // the trailing bits of a subnormal float represent
+                // the correct trailing bits of a binary16 subnormal.
+                exp = -15; // Subnormal encoding using -E_max.
+                float f_adjust = abs_f * 0x1.0p-125f;
+                signif_bits = (short)(Float.floatToRawIntBits(f_adjust) & 0x03ff);
```

Comment:
I think the `if` and the `exp++` can be avoided if the `& 0x03ff` is dropped altogether.
```
               signif_bits = (short)Float.floatToRawIntBits(f_adjust);
```
The reason is the same as for the normalized case below: a carry will eventually flow into the representation for the exponent.